### PR TITLE
PWGHF: update of DDbar correlation code (remove obsolete tasks + add process switch)

### PR DESCRIPTION
**Pull request by fcolamar**

### CODE REVIEW

File: PWGHF/Tasks/HFCorrelatorD0D0bar.cxx [hunk 1]
```diff
@@ -81,7 +68,23 @@ struct HfCorrelatorD0D0bar {
...
+     {"hSelectionStatusMCRec", "D0,D0bar candidates - MC reco;selection status;entries", {HistType::kTH1F, {{4, -0.5, 3.5}}}},
+     {"hEtaMCRec", "D0,D0bar candidates - MC reco;candidate #it{#eta};entries", {HistType::kTH1F, {{100, -5., 5.}}}},
+     {"hPhiMCRec", "D0,D0bar candidates - MC reco;candidate #it{#varphi};entries", {HistType::kTH1F, {{32, 0., 2. * o2::constants::math::PI}}}},
+     {"hYMCRec", "D0,D0bar candidates - MC reco;candidate #it{#y};entries", {HistType::kTH1F, {{100, -5., 5.}}}},
+     {"hMCEvtCount", "Event counter - MC gen;;entries", {HistType::kTH1F, {{1, -0.5, 0.5}}}},
+     {"hPtCandMCGen", "D0,D0bar particles - MC gen;particle #it{p}_{T} (GeV/#it{c});entries", {HistType::kTH1F, {{180, 0., 36.}}}},
+     {"hEtaMCGen", "D0,D0bar particles - MC gen;particle #it{#eta};entries", {HistType::kTH1F, {{100, -5., 5.}}}},
+     {"hPhiMCGen", "D0,D0bar particles - MC gen;particle #it{#varphi};entries", {HistType::kTH1F, {{32, 0., 2. * o2::constants::math::PI}}}},
+     {"hYMCGen", "D0,D0bar candidates - MC gen;candidate #it{#y};entries", {HistType::kTH1F, {{100, -5., 5.}}}},
```
Comment (vkucera) [inline PWGHF/Tasks/HFCorrelatorD0D0bar.cxx:83]:
```suggestion
     {"hYMCGen", "D0,D0bar candidates - MC gen;candidate #it{y};entries", {HistType::kTH1F, {{100, -5., 5.}}}},
```

File: PWGHF/Tasks/HFCorrelatorD0D0bar.cxx [hunk 2]
```diff
@@ -81,7 +68,23 @@ struct HfCorrelatorD0D0bar {
...
+     {"hDDbarVsEtaCut", "D0,D0bar pairs vs #eta cut;#eta_{max};entries", {HistType::kTH2F, {{(int)(maxEtaCut / incrementEtaCut), 0., maxEtaCut}, {(int)(ptThresholdForMaxEtaCut / incrementPtThreshold), 0., ptThresholdForMaxEtaCut}}}},
+     {"hPtCandMCRec", "D0,D0bar candidates - MC reco;candidate #it{p}_{T} (GeV/#it{c});entries", {HistType::kTH1F, {{180, 0., 36.}}}},
+     {"hPtProng0MCRec", "D0,D0bar candidates - MC reco;prong 0 #it{p}_{T} (GeV/#it{c});entries", {HistType::kTH1F, {{180, 0., 36.}}}},
+     {"hPtProng1MCRec", "D0,D0bar candidates - MC reco;prong 1 #it{p}_{T} (GeV/#it{c});entries", {HistType::kTH1F, {{180, 0., 36.}}}},
+     {"hSelectionStatusMCRec", "D0,D0bar candidates - MC reco;selection status;entries", {HistType::kTH1F, {{4, -0.5, 3.5}}}},
+     {"hEtaMCRec", "D0,D0bar candidates - MC reco;candidate #it{#eta};entries", {HistType::kTH1F, {{100, -5., 5.}}}},
+     {"hPhiMCRec", "D0,D0bar candidates - MC reco;candidate #it{#varphi};entries", {HistType::kTH1F, {{32, 0., 2. * o2::constants::math::PI}}}},
+     {"hYMCRec", "D0,D0bar candidates - MC reco;candidate #it{#y};entries", {HistType::kTH1F, {{100, -5., 5.}}}},
```
Comment (vkucera) [inline PWGHF/Tasks/HFCorrelatorD0D0bar.cxx:78]:
```suggestion
     {"hYMCRec", "D0,D0bar candidates - MC reco;candidate #it{y};entries", {HistType::kTH1F, {{100, -5., 5.}}}},
```

File: PWGHF/Tasks/HFCorrelatorDplusDminus.cxx [hunk 3]
```diff
@@ -62,7 +50,9 @@ const int npTBinsMassAndEfficiency = o2::analysis::hf_cuts_dplus_topikpi::npTBin
 const double efficiencyDmesonDefault[npTBinsMassAndEfficiency] = {};
 auto efficiencyDmeson_v = std::vector<double>{efficiencyDmesonDefault, efficiencyDmesonDefault + npTBinsMassAndEfficiency};
 
-/// Dplus-Dminus correlation pair builder - for real data and data-like analysis (i.e. reco-level w/o matching request via MC truth)
+using MCParticlesPlus = soa::Join<aod::McParticles, aod::HfCandProng2MCGen>;
+using MCParticlesPlus3Prong = soa::Join<aod::McParticles, aod::HfCandProng3MCGen>;
```
Comment (vkucera) [inline PWGHF/Tasks/HFCorrelatorDplusDminus.cxx:54]: Shouldn't `MCParticlesPlus` be called `MCParticlesPlus2Prong` for consistency?

File: PWGHF/Tasks/HFCorrelatorDplusDminus.cxx [hunk 4]
```diff
@@ -80,7 +70,24 @@ struct HfCorrelatorDplusDminus {
      {"hY", "Dplus,Dminus candidates;candidate #it{#y};entries", {HistType::kTH1F, {{100, -5., 5.}}}},
```
Comment (vkucera) [inline PWGHF/Tasks/HFCorrelatorDplusDminus.cxx:70]:
```suggestion
     {"hY", "Dplus,Dminus candidates;candidate #it{y};entries", {HistType::kTH1F, {{100, -5., 5.}}}},
```

File: PWGHF/Tasks/HFCorrelatorDplusDminus.cxx [hunk 5]
```diff
@@ -80,7 +70,24 @@ struct HfCorrelatorDplusDminus {
...
+     {"hPtCandMCRec", "Dplus,Dminus candidates - MC reco;candidate #it{p}_{T} (GeV/#it{c});entries", {HistType::kTH1F, {{180, 0., 36.}}}},
+     {"hPtProng0MCRec", "Dplus,Dminus candidates - MC reco;prong 0 #it{p}_{T} (GeV/#it{c});entries", {HistType::kTH1F, {{180, 0., 36.}}}},
+     {"hPtProng1MCRec", "Dplus,Dminus candidates - MC reco;prong 1 #it{p}_{T} (GeV/#it{c});entries", {HistType::kTH1F, {{180, 0., 36.}}}},
+     {"hPtProng2MCRec", "Dplus,Dminus candidates - MC reco;prong 2 #it{p}_{T} (GeV/#it{c});entries", {HistType::kTH1F, {{180, 0., 36.}}}},
+     {"hSelectionStatusMCRec", "Dplus,Dminus candidates - MC reco;selection status;entries", {HistType::kTH1F, {{2, -0.5, 1.5}}}},
+     {"hEtaMCRec", "Dplus,Dminus candidates - MC reco;candidate #it{#eta};entries", {HistType::kTH1F, {{100, -5., 5.}}}},
+     {"hPhiMCRec", "Dplus,Dminus candidates - MC reco;candidate #it{#varphi};entries", {HistType::kTH1F, {{32, 0., 2. * o2::constants::math::PI}}}},
+     {"hYMCRec", "Dplus,Dminus candidates - MC reco;candidate #it{#y};entries", {HistType::kTH1F, {{100, -5., 5.}}}},
```
Comment (vkucera) [inline PWGHF/Tasks/HFCorrelatorDplusDminus.cxx:81]:
```suggestion
     {"hYMCRec", "Dplus,Dminus candidates - MC reco;candidate #it{y};entries", {HistType::kTH1F, {{100, -5., 5.}}}},
```

File: PWGHF/Tasks/HFCorrelatorDplusDminus.cxx [hunk 6]
```diff
@@ -80,7 +70,24 @@ struct HfCorrelatorDplusDminus {
...
+     {"hMCEvtCount", "Event counter - MC gen;;entries", {HistType::kTH1F, {{1, -0.5, 0.5}}}},
+     {"hPtCandMCGen", "Dplus,Dminus particles - MC gen;particle #it{p}_{T} (GeV/#it{c});entries", {HistType::kTH1F, {{180, 0., 36.}}}},
+     {"hEtaMCGen", "Dplus,Dminus particles - MC gen;particle #it{#eta};entries", {HistType::kTH1F, {{100, -5., 5.}}}},
+     {"hPhiMCGen", "Dplus,Dminus particles - MC gen;particle #it{#varphi};entries", {HistType::kTH1F, {{32, 0., 2. * o2::constants::math::PI}}}},
+     {"hYMCGen", "Dplus,Dminus candidates - MC gen;candidate #it{#y};entries", {HistType::kTH1F, {{100, -5., 5.}}}},
```
Comment (vkucera) [inline PWGHF/Tasks/HFCorrelatorDplusDminus.cxx:86]:
```suggestion
     {"hYMCGen", "Dplus,Dminus candidates - MC gen;candidate #it{y};entries", {HistType::kTH1F, {{100, -5., 5.}}}},
```

File: PWGHF/Tasks/HFCorrelatorD0D0bar.cxx [hunk 7]
```diff
@@ -98,11 +101,20 @@ struct HfCorrelatorD0D0bar {
     registry.add("hMass", "D0,D0bar candidates;inv. mass (#pi K) (GeV/#it{c}^{2});entries", {HistType::kTH2F, {{120, 1.5848, 2.1848}, {(std::vector<double>)bins, "#it{p}_{T} (GeV/#it{c})"}}});
     registry.add("hMassD0", "D0,D0bar candidates;inv. mass D0 only (#pi K) (GeV/#it{c}^{2});entries", {HistType::kTH2F, {{120, 1.5848, 2.1848}, {(std::vector<double>)bins, "#it{p}_{T} (GeV/#it{c})"}}});
     registry.add("hMassD0bar", "D0,D0bar candidates;inv. mass D0bar only (#pi K) (GeV/#it{c}^{2});entries", {HistType::kTH2F, {{120, 1.5848, 2.1848}, {(std::vector<double>)bins, "#it{p}_{T} (GeV/#it{c})"}}});
+    registry.add("hMassD0MCRecSig", "D0 signal candidates - MC reco;inv. mass (#pi K) (GeV/#it{c}^{2});entries", {HistType::kTH2F, {{120, 1.5848, 2.1848}, {(std::vector<double>)bins, "#it{p}_{T} (GeV/#it{c})"}}});
```
Comment (vkucera) [inline PWGHF/Tasks/HFCorrelatorD0D0bar.cxx:104]: I suppose you want to define `vbins = (std::vector<double>)bins`.

**Comment by fcolamar** [comment]
Thanks Vit, I implemented your comments in the code

File: PWGHF/Tasks/HFCorrelatorD0D0bar.cxx [hunk 8]
```diff
@@ -9,7 +9,7 @@
 // granted to it by virtue of its status as an Intergovernmental Organization
 // or submit itself to any jurisdiction.
 
-/// \file HFCorrelatorD0D0bar.cxx
+/// \file HfCorrelatorD0D0bar.cxx
```
Comment (vkucera) [inline PWGHF/Tasks/HFCorrelatorD0D0bar.cxx:12]: The file name has not changed.

File: PWGHF/Tasks/HFCorrelatorD0D0bar.cxx [hunk 9]
```diff
@@ -78,10 +65,26 @@ struct HfCorrelatorD0D0bar {
...
+     {"hPtCandMCGen", "D0,D0bar particles - MC gen;particle #it{p}_{T} (GeV/#it{c});entries", {HistType::kTH1F, {{180, 0., 36.}}}},
+     {"hEtaMCGen", "D0,D0bar particles - MC gen;particle #it{#eta};entries", {HistType::kTH1F, {{100, -5., 5.}}}},
+     {"hPhiMCGen", "D0,D0bar particles - MC gen;particle #it{#varphi};entries", {HistType::kTH1F, {{32, 0., 2. * o2::constants::math::PI}}}},
+     {"hYMCGen", "D0,D0bar candidates - MC gen;candidate #it{y};entries", {HistType::kTH1F, {{100, -5., 5.}}}},
+     {"hcountD0D0barPerEvent", "D0,D0bar particles - MC gen;Number per event;entries", {HistType::kTH1F, {{20, 0., 20.}}}},
+     {"hDDbarVsDaughterEtaCut", "D0,D0bar pairs vs #eta cut on D daughters;#eta_{max};entries", {HistType::kTH2F, {{(int)(maxEtaCut / incrementEtaCut), 0., maxEtaCut}, {(int)(ptThresholdForMaxEtaCut / incrementPtThreshold), 0., ptThresholdForMaxEtaCut}}}},
+     {"hcountCCbarPerEvent", "c,cbar particles - MC gen;Number per event;entries", {HistType::kTH1F, {{20, 0., 20.}}}},
+     {"hcountCCbarPerEventPreEtaCut", "c,cbar particles - MC gen;Number per event pre #eta cut;entries", {HistType::kTH1F, {{20, 0., 20.}}}}}};
```
Comment (vkucera) [inline PWGHF/Tasks/HFCorrelatorD0D0bar.cxx:87]: Please fix the histogram names.
What is pre?
`hDDbarVsDaughterEtaCut` is missing the pT axis label.

File: PWGHF/Tasks/HFCorrelatorD0D0bar.cxx [hunk 10]
```diff
@@ -95,14 +98,24 @@ struct HfCorrelatorD0D0bar {
...
+    registry.add("hMassD0MCRecBkg", "D0 background candidates - MC reco;inv. mass (#pi K) (GeV/#it{c}^{2});entries", {HistType::kTH2F, {{120, 1.5848, 2.1848}, {vbins, "#it{p}_{T} (GeV/#it{c})"}}});
+    registry.add("hMassD0barMCRecSig", "D0bar signal candidates - MC reco;inv. mass D0bar only (#pi K) (GeV/#it{c}^{2});entries", {HistType::kTH2F, {{120, 1.5848, 2.1848}, {vbins, "#it{p}_{T} (GeV/#it{c})"}}});
+    registry.add("hMassD0barMCRecRefl", "D0bar reflection candidates - MC reco;inv. mass D0bar only (#pi K) (GeV/#it{c}^{2});entries", {HistType::kTH2F, {{120, 1.5848, 2.1848}, {vbins, "#it{p}_{T} (GeV/#it{c})"}}});
+    registry.add("hMassD0barMCRecBkg", "D0bar background candidates - MC reco;inv. mass D0bar only (#pi K) (GeV/#it{c}^{2});entries", {HistType::kTH2F, {{120, 1.5848, 2.1848}, {vbins, "#it{p}_{T} (GeV/#it{c})"}}});
+    registry.add("hcountD0triggersMCGen", "D0 trigger particles - MC gen;;N of trigger D0", {HistType::kTH2F, {{1, -0.5, 0.5}, {vbins, "#it{p}_{T} (GeV/#it{c})"}}});
+    registry.add("hcountCtriggersMCGen", "c trigger particles - MC gen;;N of trigger c quark", {HistType::kTH2F, {{1, -0.5, 0.5}, {vbins, "#it{p}_{T} (GeV/#it{c})"}}});
```
Comment (vkucera) [inline PWGHF/Tasks/HFCorrelatorD0D0bar.cxx:112]: Please fix the histogram names.

File: PWGHF/Tasks/HFCorrelatorD0D0barBarrelFullPID.cxx [hunk 11]
```diff
@@ -78,10 +65,26 @@ struct HFCorrelatorD0D0barBarrelFullPID {
...
+     {"hPtCandMCGen", "D0,D0bar particles - MC gen;particle #it{p}_{T} (GeV/#it{c});entries", {HistType::kTH1F, {{180, 0., 36.}}}},
+     {"hEtaMCGen", "D0,D0bar particles - MC gen;particle #it{#eta};entries", {HistType::kTH1F, {{100, -5., 5.}}}},
+     {"hPhiMCGen", "D0,D0bar particles - MC gen;particle #it{#varphi};entries", {HistType::kTH1F, {{32, 0., 2. * o2::constants::math::PI}}}},
+     {"hYMCGen", "D0,D0bar candidates - MC gen;candidate #it{y};entries", {HistType::kTH1F, {{100, -5., 5.}}}},
+     {"hcountD0D0barPerEvent", "D0,D0bar particles - MC gen;Number per event;entries", {HistType::kTH1F, {{20, 0., 20.}}}},
+     {"hDDbarVsDaughterEtaCut", "D0,D0bar pairs vs #eta cut on D daughters;#eta_{max};entries", {HistType::kTH2F, {{(int)(maxEtaCut / incrementEtaCut), 0., maxEtaCut}, {(int)(ptThresholdForMaxEtaCut / incrementPtThreshold), 0., ptThresholdForMaxEtaCut}}}},
+     {"hcountCCbarPerEvent", "c,cbar particles - MC gen;Number per event;entries", {HistType::kTH1F, {{20, 0., 20.}}}},
+     {"hcountCCbarPerEventPreEtaCut", "c,cbar particles - MC gen;Number per event pre #eta cut;entries", {HistType::kTH1F, {{20, 0., 20.}}}}}};
```
Comment (vkucera) [inline PWGHF/Tasks/HFCorrelatorD0D0barBarrelFullPID.cxx:87]: See above

File: PWGHF/Tasks/HFCorrelatorD0D0barBarrelFullPID.cxx [hunk 12]
```diff
@@ -95,14 +98,24 @@ struct HFCorrelatorD0D0barBarrelFullPID {
...
+    registry.add("hMassD0MCRecBkg", "D0 background candidates - MC reco;inv. mass (#pi K) (GeV/#it{c}^{2});entries", {HistType::kTH2F, {{120, 1.5848, 2.1848}, {vbins, "#it{p}_{T} (GeV/#it{c})"}}});
+    registry.add("hMassD0barMCRecSig", "D0bar signal candidates - MC reco;inv. mass D0bar only (#pi K) (GeV/#it{c}^{2});entries", {HistType::kTH2F, {{120, 1.5848, 2.1848}, {vbins, "#it{p}_{T} (GeV/#it{c})"}}});
+    registry.add("hMassD0barMCRecRefl", "D0bar reflection candidates - MC reco;inv. mass D0bar only (#pi K) (GeV/#it{c}^{2});entries", {HistType::kTH2F, {{120, 1.5848, 2.1848}, {vbins, "#it{p}_{T} (GeV/#it{c})"}}});
+    registry.add("hMassD0barMCRecBkg", "D0bar background candidates - MC reco;inv. mass D0bar only (#pi K) (GeV/#it{c}^{2});entries", {HistType::kTH2F, {{120, 1.5848, 2.1848}, {vbins, "#it{p}_{T} (GeV/#it{c})"}}});
+    registry.add("hcountD0triggersMCGen", "D0 trigger particles - MC gen;;N of trigger D0", {HistType::kTH2F, {{1, -0.5, 0.5}, {vbins, "#it{p}_{T} (GeV/#it{c})"}}});
+    registry.add("hcountCtriggersMCGen", "c trigger particles - MC gen;;N of trigger c quark", {HistType::kTH2F, {{1, -0.5, 0.5}, {vbins, "#it{p}_{T} (GeV/#it{c})"}}});
```
Comment (vkucera) [inline PWGHF/Tasks/HFCorrelatorD0D0barBarrelFullPID.cxx:112]: See above

File: PWGHF/Tasks/HFCorrelatorD0D0barBarrelFullPID.cxx [hunk 13]
```diff
@@ -836,132 +494,23 @@ struct HfCorrelatorCCbarBarrelFullPIDMCGen {
...
-      workflow.push_back(adaptAnalysisTask<HFCorrelatorD0D0barBarrelFullPIDMcRecLs>(cfgc));
-    } else if (doMCccbar) { //MC-Reco analysis
-      workflow.push_back(adaptAnalysisTask<HfCorrelatorCCbarBarrelFullPIDMCGenLs>(cfgc));
-    } else { //data analysis
-      workflow.push_back(adaptAnalysisTask<HFCorrelatorD0D0barBarrelFullPIDLs>(cfgc));
-    }
-  }
-
-  return workflow;
+  return WorkflowSpec{adaptAnalysisTask<HfCorrelatorD0D0barBarrelFullPid>(cfgc, TaskName{"hf-correlator-d0-d0bar-barrel-full-pid"})};
```
Comment (vkucera) [inline PWGHF/Tasks/HFCorrelatorD0D0barBarrelFullPID.cxx:515]: Do not use `TaskName`.

File: PWGHF/Tasks/HFCorrelatorDplusDminus.cxx [hunk 14]
```diff
@@ -77,10 +67,27 @@ struct HfCorrelatorDplusDminus {
...
+     {"hPtCandMCGen", "Dplus,Dminus particles - MC gen;particle #it{p}_{T} (GeV/#it{c});entries", {HistType::kTH1F, {{180, 0., 36.}}}},
+     {"hEtaMCGen", "Dplus,Dminus particles - MC gen;particle #it{#eta};entries", {HistType::kTH1F, {{100, -5., 5.}}}},
+     {"hPhiMCGen", "Dplus,Dminus particles - MC gen;particle #it{#varphi};entries", {HistType::kTH1F, {{32, 0., 2. * o2::constants::math::PI}}}},
+     {"hYMCGen", "Dplus,Dminus candidates - MC gen;candidate #it{y};entries", {HistType::kTH1F, {{100, -5., 5.}}}},
+     {"hcountDplusDminusPerEvent", "Dplus,Dminus particles - MC gen;Number per event;entries", {HistType::kTH1F, {{20, 0., 20.}}}},
+     {"hDDbarVsDaughterEtaCut", "Dplus,Dminus pairs vs #eta cut on D daughters;#eta_{max};entries", {HistType::kTH2F, {{(int)(maxEtaCut / incrementEtaCut), 0., maxEtaCut}, {(int)(ptThresholdForMaxEtaCut / incrementPtThreshold), 0., ptThresholdForMaxEtaCut}}}},
+     {"hcountCCbarPerEvent", "c,cbar particles - MC gen;Number per event;entries", {HistType::kTH1F, {{20, 0., 20.}}}},
+     {"hcountCCbarPerEventPreEtaCut", "c,cbar particles - MC gen;Number per event pre #eta cut;entries", {HistType::kTH1F, {{20, 0., 20.}}}}}};
```
Comment (vkucera) [inline PWGHF/Tasks/HFCorrelatorDplusDminus.cxx:90]: See above

File: PWGHF/Tasks/HFCorrelatorDplusDminus.cxx [hunk 15]
```diff
@@ -836,135 +477,26 @@ struct HfCorrelatorCCbarMcGen {
...
-      workflow.push_back(adaptAnalysisTask<HfCorrelatorDplusDminusMcRecLs>(cfgc));
-    } else if (doMCccbar) { //MC-Reco analysis
-      workflow.push_back(adaptAnalysisTask<HfCorrelatorCCbarMcGenLs>(cfgc));
-    } else { //data analysis
-      workflow.push_back(adaptAnalysisTask<HfCorrelatorDplusDminusLs>(cfgc));
-    }
-  }
-
-  return workflow;
+  return WorkflowSpec{adaptAnalysisTask<HfCorrelatorDplusDminus>(cfgc, TaskName{"hf-correlator-dplus-dminus"})};
```
Comment (vkucera) [inline PWGHF/Tasks/HFCorrelatorDplusDminus.cxx:501]: See above

File: PWGHF/Tasks/taskCorrelationDDbar.cxx [hunk 16]
```diff
@@ -513,20 +459,11 @@ struct HfTaskCorrelationDDbarMcGen {
...
-  const bool doMCRec = cfgc.options().get<bool>("doMCRec");
-  if (doMCGen) { //MC-Gen analysis
-    workflow.push_back(adaptAnalysisTask<HfTaskCorrelationDDbarMcGen>(cfgc));
-  } else if (doMCRec) { //MC-Rec analysis
-    workflow.push_back(adaptAnalysisTask<HfTaskCorrelationDDbarMcRec>(cfgc));
-  } else { //data analysis
-    workflow.push_back(adaptAnalysisTask<HfTaskCorrelationDDbar>(cfgc));
-  }
-  return workflow;
+  return WorkflowSpec{adaptAnalysisTask<HfTaskCorrelationDDbar>(cfgc, TaskName{"hf-task-correlation-d-dbar"})};
```
Comment (vkucera) [inline PWGHF/Tasks/taskCorrelationDDbar.cxx:468]: See above

File: PWGHF/Tasks/HFCorrelatorD0D0bar.cxx [hunk 17]
```diff
@@ -836,132 +494,23 @@ struct HfCorrelatorCCbarMcGen {
...
-      workflow.push_back(adaptAnalysisTask<HfCorrelatorD0D0barMcRecLs>(cfgc));
-    } else if (doMCccbar) { //MC-Reco analysis
-      workflow.push_back(adaptAnalysisTask<HfCorrelatorCCbarMcGenLs>(cfgc));
-    } else { //data analysis
-      workflow.push_back(adaptAnalysisTask<HfCorrelatorD0D0barLs>(cfgc));
-    }
-  }
-
-  return workflow;
+  return WorkflowSpec{adaptAnalysisTask<HfCorrelatorD0D0bar>(cfgc, TaskName{"hf-correlator-d0-d0bar"})};
```
Comment (vkucera) [inline PWGHF/Tasks/HFCorrelatorD0D0bar.cxx:515]: Do not use `TaskName`.

**Review by vkucera — CHANGES_REQUESTED** [review]

File: PWGHF/Tasks/HFCorrelatorD0D0bar.cxx (hunk 8 above)
Comment (fcolamar) [reply PWGHF/Tasks/HFCorrelatorD0D0bar.cxx]: Fixed

File: PWGHF/Tasks/HFCorrelatorD0D0bar.cxx (hunk 9 above)
Comment (fcolamar) [reply PWGHF/Tasks/HFCorrelatorD0D0bar.cxx]: Fixed
Changed Pre -> Before

File: PWGHF/Tasks/HFCorrelatorD0D0barBarrelFullPID.cxx (hunk 12 above)
Comment (fcolamar) [reply PWGHF/Tasks/HFCorrelatorD0D0barBarrelFullPID.cxx]: Fixed

File: PWGHF/Tasks/HFCorrelatorD0D0barBarrelFullPID.cxx (hunk 13 above)
Comment (fcolamar) [reply PWGHF/Tasks/HFCorrelatorD0D0barBarrelFullPID.cxx]: Fixed

File: PWGHF/Tasks/taskCorrelationDDbar.cxx (hunk 16 above)
Comment (fcolamar) [reply PWGHF/Tasks/taskCorrelationDDbar.cxx:468]: Fixed

File: PWGHF/Tasks/HFCorrelatorD0D0bar.cxx (hunk 17 above)
Comment (fcolamar) [reply PWGHF/Tasks/HFCorrelatorD0D0bar.cxx]: Fixed

**Comment by fcolamar** [comment]
Hello @vkucera and @ginnocen, I think that with my last commit all the comments raised were addresses, could you confirm that this is ok so this PR can be merged? Thanks!

File: PWGHF/Tasks/HFCorrelatorD0D0bar.cxx [hunk 18]
```diff
@@ -483,325 +436,30 @@ struct HfCorrelatorD0D0barMcGen {
...
-    registry.add("hcountCtriggersMCGen", "c trigger particles - MC gen;;N of trigger c quark", {HistType::kTH2F, {{1, -0.5, 0.5}, {(std::vector<double>)bins, "#it{p}_{T} (GeV/#it{c})"}}});
-  }
-
-  void process(aod::McCollision const& mccollision, MCParticlesPlus const& particlesMC)
+  /// c-cbar correlator table builder - for MC gen-level analysis
+  void processccbar(aod::McCollision const& mccollision, MCParticlesPlus const& particlesMC)
   {
     registry.fill(HIST("hMCEvtCount"), 0);
-    int counterccbar = 0, counterccbarPreEtasel = 0;
+    int counterccbar = 0, counterccbarBeforeEtasel = 0;
```
Comment (vkucera) [inline PWGHF/Tasks/HFCorrelatorD0D0bar.cxx:450]:
```suggestion
    int counterccbar = 0, counterCCbarBeforeEtaSel = 0;
```

**Comment by vkucera** [comment]
> Hello @vkucera and @ginnocen, I think that with my last commit all the comments raised were addresses, could you confirm that this is ok so this PR can be merged? Thanks!

Hi @fcolamar , sorry for taking long to come back to this PR. Thanks for the fixes. I just have one small naming suggestion.

**Comment by fcolamar** [comment]
Hi @vkucera, thanks, just fixed with the new commit, ciao!

**Comment by vkucera** [comment]
Hi @fcolamar , can you please resolve the conflicts?

**Comment by fcolamar** [comment]
Hi @vkucera, should be ok now: the conflicts were due to the McParticles_000 updates done on the previous version of the code (while the new was in the open PR). Now everything should be ok unless further modifications are made meanwhile

**Comment by vkucera** [comment]
Thanks @fcolamar for the fix.
I see that you use the following binnings at many places:
```c++
{100, -5., 5.}
{32, 0., 2. * o2::constants::math::PI}
{120, 1.5848, 2.1848}
```
I would suggest to define them once as variables and then use only the variables in the histogram declarations instead. It will save you a lot of changes if you decide to change the binning.

**Comment by jgrosseo** [comment]
Please rebase properly and then reopen

**Pull request closed without merging**